### DISCUSSION TITLE
Tableview object iid-s

### DISCUSSION THREAD

**Discussion by fromhun** [israel-dryer/ttkbootstrap#223] [Q&A] [answered]
Hi!

I'm trying to figure out how I can set my own iid-s to the tableview object. Before I used the simple treeview for this purpuse in my code but it would be nicer to use this widget. My only problem is that I can't figure out how to set the iid-s to my custom ones. It would be very helpful in my usecase, because I'm displaying database data what already has it's own uniqe id-s. 

If it's not possible I have a workaround with a hidden column which contains the database id but I don't really like this solution.

Thank you in advance for your help.

**Comment by israel-dryer** [comment, marked as answer]
Using a hidden column is going to be the only solution with the Tableview. I use the iid internally to manage the functionality of the widget.

You can use the [Tableview.hide_selected_column](https://ttkbootstrap.readthedocs.io/en/latest/api/tableview/tableview/#ttkbootstrap.tableview.Tableview.hide_selected_column) method to hide your id column.

Thinking of this has given me an idea that might make using the row data easier in the future for this type of thing... I could return the row values as a `namedtuple` so that the column headers are the field names, similar to what is done in pyodbc. So, you would be able to access the row elements using something like `row.id` or `row.first_name`, etc...

**Reply by fromhun** [reply]
Thank you for your reply. 
I already wrote the workaround, but your idea would be very usefull for future users so they can avoid complicated lines when they want to access the data from the table.

Great work anyway and thank you for it.
